StashiKu/aws-shop-be
# AWS-BE-5: Integration with S3

### link to FE deploy - https://d3ar6ba8glakkn.cloudfront.net/
### link to PR for FE part - https://github.com/StashiKu/shop-angular-cloudfront/pull/4

  ### Functionality:
 **Reviewers should verify the lambda functions by invoking them through provided URLs.**
  - `importProductsFile` - expects a request with a name of CSV file with products and creating a new Signed URL with the following key: `uploaded/${fileName}`. 
  -  `importFileParser ` - uses a readable stream to get an object from S3, parses it using csv-parser package and logs each record to be shown in CloudWatch.

 
### How to test
 1. **Import Products File**
  - [test-upload-review.csv](https://github.com/StashiKu/aws-shop-be/files/9893643/test-upload-review.csv) You can take this file if you want. It might be renamed with a specific name.
  - Upload it via admin Ui https://d3ar6ba8glakkn.cloudfront.net/admin/products
  - Ping me and I will send you screenshot of uploaded file. That's why name of the file should be specific so changes would be obvious for you
2. **importFileParser**
 - Also will send you screenshot with logs 

  ### More info:
 - `Jest` is used for testing. Run `npm run test` to run tests
 - `TypeScript`
 - `eslint`
 
**link to requirements** https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/5_integration_with_s3/task.md

### Evaluation criteria
[**1**] - File `serverless.yml` contains configuration for importProductsFile function
[**3**]- The `importProductsFile` lambda function returns a correct response which can be used to upload a file into the S3 bucket
[**4**]- Frontend application is integrated with `importProductsFile` lambda
[**5**]- The `importFileParser` lambda function is implemented and serverless.yml contains configuration for the lambda

**Additional (optional) tasks**
[**+1**] -  async/await is used in lambda functions
[**+1**] - `importProductsFile` lambda is covered by unit tests.
[**+1**] - At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into a new folder in the same bucket called parsed, and then deleted from uploaded folder)

### Total score - 8

**Thanks for reviewing** 👍 

### ⚠️ PS: if approve, please write a comment below "Approved"
